packages/ui/src/components/input-field/select-input.tsx

```diff
@@ -0,0 +1,99 @@
+import * as RadixSelect from '@radix-ui/react-select';
+import clsx from 'clsx';
+import { m } from 'framer-motion';
+import React from 'react';
+
+import { easeInOut } from '../animation';
+import { IconChevronDown, IconChevronUp, IconCheck } from '../icons';
+import { styles } from './styles';
+
+export type SelectInputProps = RadixSelect.SelectProps &
+  Pick<RadixSelect.SelectTriggerProps, 'placeholder'> & {
+    className?: string;
+    onChange?: RadixSelect.SelectProps['onValueChange'];
+  };
+
+const _SelectInput = React.forwardRef(function SelectInputComponent(
+  props: SelectInputProps,
+  forwardedRef: React.ForwardedRef<HTMLButtonElement>,
+): JSX.Element {
+  return (
+    <RadixSelect.Root
+      {...props}
+      onValueChange={props.onValueChange ?? props.onChange}
+    >
+      <RadixSelect.Trigger
+        asChild
+        className="transition duration-150 ease-in-out data-[placeholder]:font-normal data-[placeholder]:text-gray-900"
+        ref={forwardedRef}
+      >
+        <button
```

Comment:
Reuse `BaseButton`?

Reply:
I prefer using a plain button here. They only share a few styles in common, and all functionalities are handled by Radix, so not much for reuse. But thanks for pointing out `select-none` and `touch-none`.